### PR TITLE
add priorities to task

### PR DESCRIPTION
**Pull request by bryanongjx**
Now you can add priorities to tasks with format specifier "/p"

Note priorities are only accepted in the form "low" / "medium" / "high"

An example command: task add d/feed fish p/medium

Also note priorities are optional. if not specified, it will be null.
For example, the command: task add d/feed fish will have a null priority

### CODE REVIEW

**Comment by bryanongjx** [comment]
Also note that you have to reset the tasklist.json file under data as the current tasklist with no priority attribute will crash the code. once resetted, it should work fine

**Comment by bryanongjx** [comment]
im not sure where the trailing whitespace is coming from. i checked the commit and there is no whitespace. the checkstyle error came after i solved a merge conflict with the reminders branch

File: src/main/java/seedu/address/model/task/Task.java [hunk 1]
```diff
@@ -14,23 +15,39 @@ public class Task {
...
+        if (!hasPriority) {
+            throw new TaskHasNoPriorityException(this);
+        }
+        return this.priority;
+    }
+
+    public boolean hasPriority() {
+        return this.hasPriority;
+    }
+
```
Comment (bokuanT) [inline src/main/java/seedu/address/model/task/Task.java:50]: I checked this part, there seems to be ' ' whitespace on the line after the hasPriority() method i.e if you click the right arrow key it doesn't immediately go to the next line but continues on the same line on the ' ' chars. Delete these ' ' chars and it should be fine

**Review by bokuanT — CHANGES_REQUESTED** [review]

**Review by ryanchua00 — CHANGES_REQUESTED** [review]
Fix build failure in TaskFeedingReminder

**Pull request closed without merging**